sponsorship_compassion/models/contract_group.py

```diff
@@ -70,3 +70,38 @@ def _get_partner_for_contract(self, contract):
             if not contract.send_gifts_to
             else contract[contract.send_gifts_to]
         )
+
+    def _should_skip_invoice_generation(self, invoicing_date, contract=None):
+        self.ensure_one()
+
+        if contract is None:
+            return super()._should_skip_invoice_generation(invoicing_date)
+
+        search_filter = [
+            ("state", "!=", "cancel"),
+            ("invoice_date_due", "=", invoicing_date),
+            ("partner_id", "=", self.partner_id.id),
+            ("move_type", "=", "out_invoice"),
+            ("line_ids.contract_id", "=", contract.id),
+            (
+                "line_ids.product_id",
+                "in",
+                contract.product_ids.ids,
+            ),
+        ]
+
+        existing_invoices = self.env["account.move"].search_count(search_filter)
+
+        is_sub_proposal = contract.parent_id.child_id
```

Comment:
I can see a problem now. This test will be True for the whole time that the SUB proposal contract is active. However, this check is only useful at the first invoice generation of the contract. Any subsequent generation should behave normally as all other contracts. Could we therefore check that the contract has no invoice yet?
```suggestion
        is_sub_proposal = contract.parent_id.child_id and not contract.invoice_line_ids
```

Reply:
Okay, done.